app/models/concerns/pageflow/hosted_file.rb

```diff
@@ -67,7 +67,7 @@ def basename
     end
 
     def url
-      if attachment.present?
+      if ready?
         attachment.url
       end
```

Comment:
While this looks like a good idea, I'm not really sure of the implications of the change. Just like in the `thumbnail_url` case below, I'd suggest not make further changes unless they are really necessary.

Reply:
You're right, this looked like a quick fix because `ready?` is defined as `attachment.present?` a couple of lines above, but since `ready` is overwritten by processing state machines we should leave it as is for now.
Reverted in 6232dad4ea664772d1591aa1ee3937c005dd7179